// @esri/calcite-components/dist/cjs/responsive-07514287.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.9.2
 */
'use strict';

let getBreakpointsPromise;
function breakpointTokenToNumericalValue(style, tokenName) {
  return parseInt(style.getPropertyValue(tokenName));
}
/**
 * This util will return a breakpoints lookup object.
 *
 * Note that the breakpoints will be evaluated at the root and cached for reuse.
 *
 * @returns {Promise<Breakpoints>} The Breakpoints object.
 */
async function getBreakpoints() {
  if (getBreakpointsPromise) {
    return getBreakpointsPromise;
  }
  getBreakpointsPromise = new Promise((resolve) => {
    requestAnimationFrame(() => {
      const rootStyles = getComputedStyle(document.body);
      resolve({
        width: {
          large: breakpointTokenToNumericalValue(rootStyles, "--calcite-app-breakpoint-width-lg"),
          medium: breakpointTokenToNumericalValue(rootStyles, "--calcite-app-breakpoint-width-md"),
          small: breakpointTokenToNumericalValue(rootStyles, "--calcite-app-breakpoint-width-sm"),
          xsmall: breakpointTokenToNumericalValue(rootStyles, "--calcite-app-breakpoint-width-xs"),
        },
      });
    });
  });
  return getBreakpointsPromise;
}

exports.getBreakpoints = getBreakpoints;
