// @esri/calcite-components/dist/esm/en-ca-39c1e950.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.9.2
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-6e486031.js';
import './index-57f4809d.js';
import './form-a9c4cf79.js';
import './dom-158133b6.js';
import './guid-a5b8c53d.js';
import './resources-19304982.js';
import './interactive-ede14c07.js';
import './browser-394e2d3d.js';
import './key-cc1e5093.js';
import './label-3766b951.js';
import './loadable-46e63135.js';
import './locale-bad69fe5.js';
import './observers-72fa7eae.js';
import './focusTrapComponent-c627d9d8.js';
import './math-4bba1217.js';
import './t9n-c5cd39db.js';
import './openCloseComponent-7b0b4b4f.js';

// English (Canada) [en-ca]
var locale = {
  name: 'en-ca',
  weekdays: 'Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday'.split('_'),
  months: 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_'),
  weekdaysShort: 'Sun_Mon_Tue_Wed_Thu_Fri_Sat'.split('_'),
  monthsShort: 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_'),
  weekdaysMin: 'Su_Mo_Tu_We_Th_Fr_Sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'h:mm A',
    LTS: 'h:mm:ss A',
    L: 'YYYY-MM-DD',
    LL: 'MMMM D, YYYY',
    LLL: 'MMMM D, YYYY h:mm A',
    LLLL: 'dddd, MMMM D, YYYY h:mm A'
  },
  relativeTime: {
    future: 'in %s',
    past: '%s ago',
    s: 'a few seconds',
    m: 'a minute',
    mm: '%d minutes',
    h: 'an hour',
    hh: '%d hours',
    d: 'a day',
    dd: '%d days',
    M: 'a month',
    MM: '%d months',
    y: 'a year',
    yy: '%d years'
  }
};
dayjs.locale(locale, null, true);

export default locale;
