// @esri/calcite-components/dist/esm/it-ch-cebab7fd.js
/*!
 * All material copyright ESRI, All Rights Reserved, unless otherwise specified.
 * See https://github.com/Esri/calcite-design-system/blob/main/LICENSE.md for details.
 * v1.9.2
 */
import { d as dayjs } from './calcite-input-time-picker.calcite-time-picker-6e486031.js';
import './index-57f4809d.js';
import './form-a9c4cf79.js';
import './dom-158133b6.js';
import './guid-a5b8c53d.js';
import './resources-19304982.js';
import './interactive-ede14c07.js';
import './browser-394e2d3d.js';
import './key-cc1e5093.js';
import './label-3766b951.js';
import './loadable-46e63135.js';
import './locale-bad69fe5.js';
import './observers-72fa7eae.js';
import './focusTrapComponent-c627d9d8.js';
import './math-4bba1217.js';
import './t9n-c5cd39db.js';
import './openCloseComponent-7b0b4b4f.js';

// Italian (Switzerland) [it-ch]
var locale = {
  name: 'it-ch',
  weekdays: 'domenica_lunedì_martedì_mercoledì_giovedì_venerdì_sabato'.split('_'),
  months: 'gennaio_febbraio_marzo_aprile_maggio_giugno_luglio_agosto_settembre_ottobre_novembre_dicembre'.split('_'),
  weekStart: 1,
  weekdaysShort: 'dom_lun_mar_mer_gio_ven_sab'.split('_'),
  monthsShort: 'gen_feb_mar_apr_mag_giu_lug_ago_set_ott_nov_dic'.split('_'),
  weekdaysMin: 'do_lu_ma_me_gi_ve_sa'.split('_'),
  ordinal: function ordinal(n) {
    return n;
  },
  formats: {
    LT: 'HH:mm',
    LTS: 'HH:mm:ss',
    L: 'DD.MM.YYYY',
    LL: 'D MMMM YYYY',
    LLL: 'D MMMM YYYY HH:mm',
    LLLL: 'dddd D MMMM YYYY HH:mm'
  },
  relativeTime: {
    future: 'tra %s',
    past: '%s fa',
    s: 'alcuni secondi',
    m: 'un minuto',
    mm: '%d minuti',
    h: 'un\'ora',
    hh: '%d ore',
    d: 'un giorno',
    dd: '%d giorni',
    M: 'un mese',
    MM: '%d mesi',
    y: 'un anno',
    yy: '%d anni'
  }
};
dayjs.locale(locale, null, true);

export default locale;
